Source/WebCore/page/PointerLockController.cpp

```diff
@@ -40,44 +42,93 @@
 
 namespace WebCore {
 
+class UnadjustedMovementPlatformMouseEvent : public PlatformMouseEvent {
+public:
+    UnadjustedMovementPlatformMouseEvent(const PlatformMouseEvent& src)
+        : PlatformMouseEvent(src)
+    {
+        m_movementDelta = src.unadjustedMovementDelta();
+    }
+};
+
 PointerLockController::PointerLockController(Page& page)
     : m_page(page)
 {
 }
 
-void PointerLockController::requestPointerLock(Element* target)
+PointerLockController::~PointerLockController()
+{
+}
+
+void PointerLockController::requestPointerLock(Element* target, std::optional<PointerLockOptions>&& options, RefPtr<DeferredPromise> promise)
 {
     if (!target || !target->isConnected() || m_documentOfRemovedElementWhileWaitingForUnlock) {
         enqueueEvent(eventNames().pointerlockerrorEvent, target);
+        if (promise)
+            promise->reject(WrongDocumentError);
         return;
     }
 
     if (m_documentAllowedToRelockWithoutUserGesture != &target->document() && !UserGestureIndicator::processingUserGesture()) {
         enqueueEvent(eventNames().pointerlockerrorEvent, target);
+        // If the request was not started from an engagement gesture and the Document has not previously released a successful Pointer Lock with exitPointerLock():
+        if (promise)
+            promise->reject(NotAllowedError);
```

Comment:
Please provide a string to explain why this got rejected `{ NotAllowedError, "explanation"_s }`